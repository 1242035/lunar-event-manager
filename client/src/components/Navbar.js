import React from 'react';

/**
 * Component for ceating navbars
 * @param{props} - passed properties
 * @return{React.Component}
 */
const Navbar = props => (
  <nav className="io-navbar io-fixed-top">{props.children}</nav>
);

export default Navbar;
