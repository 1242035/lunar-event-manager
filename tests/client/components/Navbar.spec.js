import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../../client/src/components/Navbar';

describe('Navbar', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<Navbar />);
    expect(wrapper.length).toEqual(1);
  });
});
