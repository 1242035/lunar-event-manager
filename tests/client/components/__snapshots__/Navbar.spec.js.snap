// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "io-navbar io-fixed-top",
    },
    "ref": null,
    "rendered": null,
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "io-navbar io-fixed-top",
      },
      "ref": null,
      "rendered": null,
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
